src/measure_theory/function/strongly_measurable.lean

```diff
@@ -431,9 +956,611 @@ lemma ae_fin_strongly_measurable_zero {α β} {m : measurable_space α} (μ : me
+lemma _root_.continuous.comp_ae_strongly_measurable {g : β → γ} {f : α → β}
+  (hg : continuous g) (hf : ae_strongly_measurable f μ) :
+  ae_strongly_measurable (λ x, g (f x)) μ :=
+⟨_, hg.comp_strongly_measurable hf.strongly_measurable_mk, eventually_eq.fun_comp hf.ae_eq_mk g⟩
+
+/-- A continuous function from `α` to `β` is ae strongly measurable when one of the two spaces is
+second countable. -/
+lemma _root_.continuous.ae_strongly_measurable [topological_space α] [opens_measurable_space α]
+  [metrizable_space β] [second_countable_topology_either α β] (hf : continuous f) :
+  ae_strongly_measurable f μ :=
+hf.strongly_measurable.ae_strongly_measurable
+
+protected lemma prod_mk {f : α → β} {g : α → γ}
+  (hf : ae_strongly_measurable f μ) (hg : ae_strongly_measurable g μ) :
+  ae_strongly_measurable (λ x, (f x, g x)) μ :=
+⟨λ x, (hf.mk f x, hg.mk g x), hf.strongly_measurable_mk.prod_mk hg.strongly_measurable_mk,
+  hf.ae_eq_mk.prod_mk hg.ae_eq_mk⟩
+
+/-- In a space with second countable topology, measurable implies ae strongly measurable. -/
+lemma _root_.measurable.ae_strongly_measurable {m : measurable_space α}
+  {μ : measure α} [measurable_space β] [metrizable_space β]
+  [second_countable_topology β] [opens_measurable_space β] (hf : measurable f) :
+  ae_strongly_measurable f μ :=
+hf.strongly_measurable.ae_strongly_measurable
+
+section arithmetic
+
+@[to_additive]
+protected lemma mul [has_mul β] [has_continuous_mul β]
+  (hf : ae_strongly_measurable f μ) (hg : ae_strongly_measurable g μ) :
+  ae_strongly_measurable (f * g) μ :=
+⟨hf.mk f * hg.mk g, hf.strongly_measurable_mk.mul hg.strongly_measurable_mk,
+  hf.ae_eq_mk.mul hg.ae_eq_mk⟩
+
+@[to_additive]
+protected lemma mul_const [has_mul β] [has_continuous_mul β]
+  (hf : ae_strongly_measurable f μ) (c : β) :
+  ae_strongly_measurable (λ x, f x * c) μ :=
+hf.mul ae_strongly_measurable_const
+
+@[to_additive]
+protected lemma const_mul [has_mul β] [has_continuous_mul β]
+  (hf : ae_strongly_measurable f μ) (c : β) :
+  ae_strongly_measurable (λ x, c * f x) μ :=
+ae_strongly_measurable_const.mul hf
+
+@[to_additive]
+protected lemma inv [group β] [topological_group β] (hf : ae_strongly_measurable f μ) :
+  ae_strongly_measurable (f⁻¹) μ :=
+⟨(hf.mk f)⁻¹, hf.strongly_measurable_mk.inv, hf.ae_eq_mk.inv⟩
+
+@[to_additive]
+protected lemma div [group β] [topological_group β]
+  (hf : ae_strongly_measurable f μ) (hg : ae_strongly_measurable g μ) :
+  ae_strongly_measurable (f / g) μ :=
+⟨hf.mk f / hg.mk g, hf.strongly_measurable_mk.div hg.strongly_measurable_mk,
+  hf.ae_eq_mk.div hg.ae_eq_mk⟩
+
+@[to_additive]
+protected lemma smul {𝕜} [topological_space 𝕜] [has_scalar 𝕜 β] [has_continuous_smul 𝕜 β]
+  {f : α → 𝕜} {g : α → β} (hf : ae_strongly_measurable f μ) (hg : ae_strongly_measurable g μ) :
+  ae_strongly_measurable (λ x, f x • g x) μ :=
+continuous_smul.comp_ae_strongly_measurable (hf.prod_mk hg)
+
+protected lemma const_smul {𝕜} [has_scalar 𝕜 β] [has_continuous_const_smul 𝕜 β]
```

Comment:
Does `@[to_additive]` work on this?

Reply:
No, it doesn't. I had tried to make it work, but I couldn't get to the bottom of it so I just left it like that.